core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncRequestBody.java

```diff
@@ -271,34 +274,22 @@ private void closeFile() {
             }
         }
 
-        private void signalOnNext(ByteBuffer bb) {
-            synchronized (this) {
```

Comment:
`synchronized` is removed because it doesn't seem to be necessary here imo.

Reply:
Technically I think this is still needed. onError and onNext must be called serially, so the synchronized made sure that it would always happen.

Without the synchronized, we may get onError and onNext in parallel, e.g. if someone calls subscription.request(-1) while onNext is in progress.

Reply:
Yeah, that's fair.